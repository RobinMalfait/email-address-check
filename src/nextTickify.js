export default function nextTickify(fn) {
  return (...args) => {
    return new Promise((resolve, reject) => {
      process.nextTick(async () => {
        try {
          const result = await fn(...args);
          resolve(result);
        } catch (err) {
          reject(err);
        }
      });
    });
  };
}
