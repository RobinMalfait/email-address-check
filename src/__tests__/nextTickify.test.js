import nextTickify from "../nextTickify";

describe("nextTickify", () => {
  it("should resolve like a promise", () => {
    const myFn = value => value;
    const myFnNextTickified = nextTickify(myFn);

    return expect(myFnNextTickified(123)).resolves.toMatchSnapshot();
  });

  it("should reject like a promise", () => {
    const myFn = value => {
      throw new Error(value);
    };
    const myFnNextTickified = nextTickify(myFn);

    return expect(
      myFnNextTickified("this should reject")
    ).rejects.toMatchSnapshot();
  });
});
