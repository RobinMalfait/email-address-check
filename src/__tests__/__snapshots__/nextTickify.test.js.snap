// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nextTickify should reject like a promise 1`] = `[Error: this should reject]`;

exports[`nextTickify should resolve like a promise 1`] = `123`;
